controllers/storagecluster/external_resources_test.go

```diff
@@ -557,3 +557,61 @@ func TestExternalMonitoringResources(t *testing.T) {
 		})
 	}
 }
+
+func TestErasureCodedExternalResources(t *testing.T) {
+	cr := &api.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ocsinit",
+			Namespace: "openshift-storage",
+		},
+		Spec: api.StorageClusterSpec{
+			ExternalStorage: api.ExternalStorageClusterSpec{
+				Enable: true,
+			},
+			Monitoring: &api.MonitoringSpec{
+				ReconcileStrategy: string(ReconcileStrategyIgnore),
+			},
+		},
+	}
+	externalResource := []ExternalResource{
+		{
+			Name: "ceph-rbd",
+			Kind: "StorageClass",
+			Data: map[string]string{
+				"pool": generateNameForCephBlockPool(cr),
+			},
+		},
+		{
+			Name: "ceph-rbd-ec",
+			Kind: "StorageClass",
+			Data: map[string]string{
+				"dataPool": generateNameForCephBlockPool(cr),
```

Comment:
```suggestion
				"dataPool": "my-pool",
```

`generateNameForCephBlockPool()` is overriden when the cluster is external with the actual external pool name

Reply:
can you point in where it is overriden?

Reply:
https://github.com/red-hat-storage/ocs-operator/pull/1440#discussion_r772957395

Reply:
updated.